### DISCUSSION TITLE
UNNEST while preserving order?

### DISCUSSION THREAD

**Discussion by gotdan** [duckdb/duckdb#8897] [Q&A]
I have data that contains a nested array in a struct. When I unnest in the query's field list, it returns the data in the order of the source array. Here's a simplified example:
```
WITH data AS (
	SELECT {
		'nested_1': [{ 'display': 'A'}, {'display': 'B'}, {'display': 'C'}]
	} field_1
)
SELECT 
UNNEST(field_1.nested_1).display AS display
FROM data;

-- returns rows as 'A', 'B', 'C'
```

However, when I unnest in a join, the query returns the rows in a different order than the source array:
```
WITH data AS (
	SELECT {
		'nested_1': [{ 'display': 'A'}, {'display': 'B'}, {'display': 'C'}]
	} field_1
)
SELECT unnested.display
FROM data,
UNNEST(field_1.nested_1) AS t(unnested);

-- returns rows as 'C', 'B', 'A'
```

I need to be able to left join to the unnest since not all rows will have the nested array, so I think I have to do it in the join section rather than the field section of the query. Is there a way to retain the original ordering for this? In Postgres, I could unnest "WITH ORDINALITY" to get a sort index, but this doesn't look to be supported in DuckDB.

Thanks!

**Comment by nsiicm0** [comment]
Looks like `with ordinality` is in the works: https://github.com/duckdb/duckdb/pull/9014
